contracts/libraries/HyperdriveExecution.sol

```diff
@@ -0,0 +1,578 @@
+                self,
+                _asBase ? self.convertToShares(_amount) : _amount
+            );
+    }
+
+    /// @dev Calculates the number of bonds a user will receive when opening a
+    ///      long position.
+    /// @param _shareAmount Amount of shares being exchanged for bonds.
+    /// @return Amount of bonds received.
+    function _calculateOpenLong(
+        IHyperdrive self,
+        uint256 _shareAmount
+    ) internal view returns (uint256) {
+        // We must load the entire PoolConfig since it contains values from
+        // immutables without public accessors.
+        IHyperdrive.PoolConfig memory poolConfig = self.getPoolConfig();
+
+        // Prepare the necessary information to perform the yield space
+        // calculation. We save gas by reading storage directly instead of
+        // retrieving entire PoolInfo struct.
+        uint256[] memory slots = new uint256[](2);
+        slots[0] = HYPERDRIVE_SHARE_RESERVES_BOND_RESERVES_SLOT;
+        slots[1] = HYPERDRIVE_SHARE_ADJUSTMENT_SHORTS_OUTSTANDING_SLOT;
+        bytes32[] memory values = self.load(slots);
+        uint256 effectiveShareReserves = HyperdriveMath
+            .calculateEffectiveShareReserves(
+                uint128(values[0].extract_32_16(16)), // shareReserves
+                uint256(uint128(values[1].extract_32_16(16))).toInt256() // shareAdjustment
+            );
+        uint256 bondReserves = uint128(values[0].extract_32_16(0));
+        uint256 _vaultSharePrice = vaultSharePrice(self);
+
+        // Calculate the change in hyperdrive's bond reserves given a purchase
+        // of _shareAmount. This amount is equivalent to the amount of bonds
+        // the purchaser will receive (not accounting for fees).
+        uint256 bondReservesDelta = YieldSpaceMath
+            .calculateBondsOutGivenSharesInDown(
+                effectiveShareReserves,
+                bondReserves,
+                _shareAmount,
+                // NOTE: Since the bonds traded on the curve are newly minted,
+                // we use a time remaining of 1. This means that we can use
+                // `_timeStretch = t * _timeStretch`.
+                1e18 - poolConfig.timeStretch,
+                _vaultSharePrice,
+                poolConfig.initialVaultSharePrice
+            );
+
+        // Apply fees to the output bond amount and return it.
+        uint256 spotPrice = HyperdriveMath.calculateSpotPrice(
+            effectiveShareReserves,
+            bondReserves,
+            poolConfig.initialVaultSharePrice,
+            poolConfig.timeStretch
+        );
+        (, bondReservesDelta, ) = _calculateOpenLongFees(
+            _shareAmount,
+            bondReservesDelta,
+            _vaultSharePrice,
+            spotPrice,
+            poolConfig.fees.curve,
+            poolConfig.fees.governanceLP
+        );
+        return bondReservesDelta;
+    }
+
+    /// @dev Calculate the fees involved with opening the long and apply them.
```

Comment:
This looks great! One thought is that if we're not using the other return values (`shareReservesDelta` and `totalGovernanceFee`), we can probably remove those. Wdyt?